### PR TITLE
Add loading state to modal form buttons

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- Updates `dynamic-form.js` to add a loading state and disable the submit button on form submission

## QA

- Go to https://ubuntu-com-13325.demos.haus/ and find any form, fill in and submit it
- See that the button has the [vanilla state](https://vanillaframework.io/docs/patterns/buttons) and is disabled
- Repeat for different forms across u.com

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-6608

### CODE REVIEW

**Review by mtruj013 — COMMENTED** [review]
Looks awesome! The single-page modals (at least the ones I tested) QA perfectly, but the multi-page ones seem to have a bug, eg:   https://ubuntu-com-13325.demos.haus/cloud/public-cloud#get-in-touch it looks like this 
Screenshot from 2023-11-02 17-33-42

Same for https://ubuntu-com-13325.demos.haus/core/services#get-in-touch

**Comment by petesfrench** [comment]
Thanks @mtruj013 ! Can you take another look?

File: static/js/src/dynamic-forms.js [hunk 1]
```diff
@@ -541,6 +541,26 @@ import setupIntlTelInput from "./intlTelInput.js";
 
       setpreferredLanguage();
 
+      // Disables submit button and adds visual queue when it is submitted
+      function setupSubmitButton() {
+        console.log("setupSubmitButton");
```
Comment (mtruj013) [inline static/js/src/dynamic-forms.js:546]:
```suggestion

```

File: static/js/src/dynamic-forms.js [hunk 2]
```diff
@@ -541,6 +541,26 @@ import setupIntlTelInput from "./intlTelInput.js";
...
+        const modalForm = formContainer.querySelector("form");
+        const spinnerIcon = document.createElement("i");
+        spinnerIcon.className = "p-icon--spinner u-animation--spin is-light";
+        console.log("spinnerIcon", spinnerIcon);
+        modalForm.addEventListener("submit", function (e) {
+          const buttonRect = submitButton.getBoundingClientRect();
+          console.log("submit event", submitButton);
```
Comment (mtruj013) [inline static/js/src/dynamic-forms.js:553]:
```suggestion
```

**Review by mtruj013 — APPROVED** [review]
LGTM! Just some left over console logs

**Comment by petesfrench** [comment]
thanks embarrassing, thanks @mtruj013 !

**Pull request merged**